nuxeo-runtime/nuxeo-connect-standalone/src/main/java/org/nuxeo/launcher/connect/ConnectBroker.java

```diff
@@ -834,14 +873,43 @@ public boolean pkgInstall(List<String> packageIdsToInstall) {
      */
     public boolean pkgInstall(List<String> packageIdsToInstall, boolean ignoreMissing) {
         log.debug("Installing: " + packageIdsToInstall);
-        for (String pkgId : packageIdsToInstall) {
+        Queue<String> remaining = new LinkedList<>(packageIdsToInstall);
+        while (!remaining.isEmpty()) {
+            String pkgId = remaining.poll();
             if (pkgInstall(pkgId, ignoreMissing) == null && !ignoreMissing) {
                 return false;
             }
+            if (isRestartRequired()) {
+                persistCommand(serializeInstallCmd(remaining));
+                throw new LauncherRestartException();
+            }
         }
         return true;
     }
 
+    /**
+     * Persists the pending package operation into file system. It's useful when Nuxeo launcher is about to exit. Empty
+     * command line won't be persisted.
+     * <p>
+     * The given command will be appended as a new line into target file {@link #pendingFile}. NOTE: the command line
+     * options are not serialized. Therefore, they should be provided by nuxeoctl again after launcher's restart.
+     *
+     * @param command command to persist (appended as a new line)
+     * @throws IllegalStateException if any exception occurs
+     * @see #pendingFile
+     * @since 10.2
+     */
+    protected void persistCommand(String command) {
+        if (command.isEmpty()) {
+            return;
+        }
+        try {
+            Files.write(pendingFile, Arrays.asList(command), StandardOpenOption.CREATE, StandardOpenOption.APPEND);
```

Comment:
Could you replace `Arrays.asList` by `Collections.singletonList` ?

Reply:
I know that @efge prefers `Arrays#asList` which is shorter. Personally I prefer `Collections#singletonList`—but I guess we're all waiting for the `List#of` in Java 9 :)